lib/application/pages/home/collections/collections_page.dart

```diff
@@ -91,3 +95,51 @@ class CollectionsContents extends HookWidget {
     return widget.withSymmetricalPadding(context, horizontal: Spacing.medium);
   }
 }
+
+/// [CollectionsPage] empty state, given the current [collectionsVM] state.
+class _CollectionsEmptyState extends HookWidget {
+  const _CollectionsEmptyState(this.onUpdateSegment, {required this.currentSegment});
+
+  final Function(CollectionsSegment segment) onUpdateSegment;
+  final CollectionsSegment currentSegment;
+
+  @override
+  Widget build(BuildContext context) {
+    final textTheme = Theme.of(context).textTheme;
+    final theme = useTheme();
+
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        Image.asset(
+          images.folderIllustrationAsset,
+          height: dimens.collectionsEmptyStateImageSize,
+          width: dimens.collectionsEmptyStateImageSize,
+          fit: BoxFit.fill,
+          color: theme.neutralSwatch.shade700,
+        ),
+        context.verticalBox(Spacing.xLarge),
+        Text(
+          strings.collectionsEmptyTitleSegment(currentSegment),
+          style: textTheme.headline6,
+          textAlign: TextAlign.center,
+        ),
+        context.verticalBox(Spacing.medium),
+        Text(
+          strings.collectionsEmptyMessageSegment(currentSegment),
+          style: textTheme.bodyText2?.copyWith(color: theme.neutralSwatch.shade400),
+          textAlign: TextAlign.center,
+        ),
+        context.verticalBox(Spacing.xLarge),
+        ElevatedButton(
+          onPressed: () {
+            final oppositeSegment =
+                currentSegment == CollectionsSegment.explore ? CollectionsSegment.review : CollectionsSegment.explore;
+            onUpdateSegment(oppositeSegment);
```

Comment:
Not sure if this is the best solution. Open to suggestions

Reply:
I would remove nested check here, as it can lead to future problems if new tabs are added/updated.

My suggestion is to delegate its responsibility to the caller, something like:

```dart
//...
    child: _CollectionsContents(
      onSegmentSwapRequested: () {
   final nextIndex = segment.index + 1 % availableSegments.length;
   collectionsTabController.animateTo(nextIndex);
},
    ),
//

//...
class _CollectionsContents extends HookWidget {
  const _CollectionsContents({this.onSegmentSwapRequested});
  
  /// {@macro...}
  final VoidCallback onSegmentSwapRequested;
//...
}

//...
class _CollectionsEmptyState extends HookWidget {
  const _CollectionsEmptyState({required this.onSegmentSwapRequested, required this.title, required this.description});
  
  /// {@template}
  /// An empty-state request for a segment swap.
  /// ....
  final VoidCallback onSegmentSwapRequested;
 //...
```